internal/clients/gitdiff/gitdiff.go

```diff
@@ -0,0 +1,66 @@
+package gitdiff
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+const unknownCommitHash = "0000000000000000000000000000000000000000"
+
+// GitDiff client for getting diffs from the command line
+type GitDiff struct {
+	WorkDir    string
+	BaseBranch string
+	BaseSHA    string
+	Head       string
+}
+
+// GetDiff uses the command line to compute the diff
+func (gd *GitDiff) GetDiff() (string, error) {
+	err := os.Chdir(gd.WorkDir)
+	if err != nil {
+		return "", err
+	}
+
+	var diffCmd *exec.Cmd
+	if gd.BaseBranch != "" {
+		err = exec.Command("git", "fetch", "origin", gd.BaseBranch).Run()
+		if err != nil {
+			return "", err
+		}
+		diffCmd = exec.Command("git", "diff", fmt.Sprintf("origin/%s", gd.BaseBranch))
```

Comment:
Command `git diff origin/${base branch}`. The checkout action checks out the branch for which the PR is being made for (branch with code changes).

Not sure why but `git diff origin/${base branch} ${SHA}` would fail with `128` error code

Reply:
Comparing this to the bash script, does using fetch depth=1 also cause this error?

Reply:
Good point, when fetching with `--depth=1` it still works, this probably better for performance instead of fetching the entire branch.